src/sdk/pynni/nni/compression/torch/pruning/structured_pruning.py

```diff
@@ -324,19 +618,30 @@ class ActivationMeanRankFilterPrunerMasker(ActivationFilterPrunerMasker):
     "Pruning Convolutional Neural Networks for Resource Efficient Inference", ICLR 2017.
     https://arxiv.org/abs/1611.06440
     """
-    def get_mask(self, base_mask, weight, num_prune, wrapper, wrapper_idx):
-        assert wrapper_idx is not None
-        activations = self.pruner.collected_activation[wrapper_idx]
-        if len(activations) < self.statistics_batch_num:
+
+    def get_mask(self, base_mask, weight, num_prune, wrapper, wrapper_idx, channel_masks=None):
+
+        mean_activation = self.get_channel_sum(wrapper, wrapper_idx)
+        if mean_activation is None:
+            # the collected activation is not enough
             return None
-        mean_activation = self._cal_mean_activation(activations)
+        if channel_masks is not None:
```

Comment:
Let's remove this special handing for mean activation as discussed.

Reply:
Updated~